// react-jplaylist/lib/components/remove/remove.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _constants = require('../../util/constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Remove = function Remove(_ref) {
  var remove = _ref.remove,
      id = _ref.id,
      index = _ref.index,
      children = _ref.children,
      attributes = _objectWithoutProperties(_ref, ['remove', 'id', 'index', 'children']);

  return _react2.default.createElement(
    'button',
    _extends({
      className: _constants.classes.REMOVE, onClick: function onClick() {
        return remove(id, index);
      }
    }, attributes),
    children
  );
};

Remove.defaultProps = {
  children: '×'
};

Remove.propTypes = {
  children: _propTypes2.default.node,
  id: _propTypes2.default.string.isRequired,
  index: _propTypes2.default.number.isRequired,
  remove: _propTypes2.default.func.isRequired
};

exports.default = Remove;
