// react-jplaylist/lib/util/mockData/mockJPlaylistOptions.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _shortid = require('shortid');

var _shortid2 = _interopRequireDefault(_shortid);

var _constants = require('../constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _extends({}, _constants.defaultStatus, _constants.defaultOptions, {
  playlist: [{
    title: 'Bubble',
    artist: 'Miaow',
    sources: {
      m4a: 'http://jplayer.org/audio/m4a/Miaow-07-Bubble.m4a',
      oga: 'http://jplayer.org/audio/ogg/Miaow-07-Bubble.ogg'
    },
    free: true,
    id: _shortid2.default.generate(),
    poster: 'downinthevalley-poster.jpg'
  }, {
    title: 'Tempered Song',
    artist: 'Miaow',
    sources: {
      mp3: 'http://www.jplayer.org/audio/mp3/Miaow-01-Tempered-song.mp3',
      oga: 'http://www.jplayer.org/audio/ogg/Miaow-01-Tempered-song.ogg'
    },
    id: _shortid2.default.generate()
  }, {
    title: 'Cro Magnon Man',
    artist: 'The Stark Palace',
    sources: {
      mp3: 'http://www.jplayer.org/audio/mp3/TSP-01-Cro_magnon_man.mp3',
      oga: 'http://www.jplayer.org/audio/ogg/TSP-01-Cro_magnon_man.ogg'
    },
    id: _shortid2.default.generate()
  }]
});
